Tests/SwiftSyntaxBuilderTest/BreakStmtSyntaxTests.swift

```diff
@@ -17,7 +17,7 @@ import SwiftSyntaxBuilder
 final class BreakStmtSyntaxTests: XCTestCase {
   func testBreakStmtSyntax() {
     let testCases: [UInt: (StmtSyntax, String)] = [
-      #line: (BreakStmtSyntax().as(StmtSyntax.self)!, "break"),
+      #line: (StmtSyntax(BreakStmtSyntax()), "break"),
```

Comment:
Hmm, we should support `BreakStmtSyntax().as(StmtSyntax.self)`. It definitely shouldn’t emit a warning that the cast will always fail because that’s not true, it will always succeed.

I think in `LeafDeclSyntaxNodeProtocol`, we need the following (same for `is` and `cast`).

```swift
  func `as`(_ syntaxType: DeclSyntax.Type) -> Self? {
    return self
  }
```

Reply:
I've been thinking more about producing a warning when the user is trying to case to the base node with a message like:
```
 `as` method should be used to downcasting for upcasting to base node use `BaseNode.init`
```
What do you think?

Reply:
What about making it slightly more action-focused

```
Use `BaseNode.init` for upcasting.
```

Reply:
I've added such a feature for all base nodes and `Syntax` node here: https://github.com/apple/swift-syntax/pull/2108/commits/9986d57dfd80800c30b686ae05b55f8739c168e9